nixos/modules/services/audio/mpd.nix

```diff
@@ -126,49 +117,136 @@ in
         '';
       };
 
-      user = lib.mkOption {
-        type = lib.types.str;
-        default = name;
-        description = "User account under which MPD runs.";
+      openFirewall = lib.mkOption {
+        type = lib.types.bool;
+        default = lib.elem cfg.settings.bind_to_address [
```

Comment:
I'm not entirely sure I'm sold on this idea. I think there is some value in having the user explicitly say that the service should be let through the firewall.

Considering the default address is `127.0.0.1` and that most machines running mpd likely will be behind a home router, it's not a big deal. This is probably not the first module to do this either. But I still find it a bit spooky that I could accidentally open the firewall through misconfiguration seemingly unrelated to the firewall.

Reply:
> But I still find it a bit spooky that I could accidentally open the firewall through misconfiguration seemingly unrelated to the firewall.

I agree, and I have 2 suggestions:

- Warn about this in the `bind_to_address` documentation.
- Open the firewall only if `bind_to_address == "any"`.

What do you think?